### DISCUSSION TITLE
欢迎来到 Vant 讨论区 / Welcome to Vant Discussions!

### DISCUSSION THREAD

**Discussion by chenjiahan** [youzan/vant#7710] [General]
## 👋 欢迎！

   我们正在使用“讨论”作为与社区其他成员联系的地方。 在这里你可以：
   * 提出你在使用 Vant 过程中遇到的问题。
   * 分享你对 Vant 的反馈和建议。
   * 与社区的其他成员互动。

## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.

## Tips

注意：请勿在此会话下反馈 Bug 和 Feature，建议通过 issue 进行反馈。

**Comment by QqinzzzZ** [comment]
您好，在使用van-picker 渲染级联时发现当数据children 为空时切换 下一级级联不会同步的置空，为解决这一问题；采用了使用change方法动态清空，再重新赋值下一级级联，但是此时源码报错"TypeError: Cannot read property 'text' of undefined"，错误位置在onChange: function onChange(columnIndex)方法中 ，希望您可以帮忙看一下这部分报错的原因，感谢

**Reply by nbollard4** [reply]
Hello hello

**Comment by decadez** [comment]
在微信内核浏览器中使用vant。当全屏打开视频时，noticeBar组件的文字字号变大了～求解决～

**Reply by xudehong123** [reply]
目前没有提供 CDN 形式的语言包，可以手动拷贝语言包的内容来使用。 请问如何手动拷贝语言包

**Comment by leostrong8** [comment]
hello,everyone.问题describe：使用van-form里面van-filed,框架使用的vue,代码如下演示
ruleForm{
customName__c: '到时候有接口给这个属性赋值，而且赋值是成功的。'
}

>  <van-field

>            disabled

>            v-model="ruleForm.customName__c"

>            label="客户姓名"

>            placeholder="请输入客户姓名"/>

前端页面展示出来这个表单是空值，没有值。而且敲重点，这个问题只发生在ios14+版本，安卓以及ios14以下的版本都没有这样的问题。
修改后：

> <van-field

>               disabled

>                :value="ruleForm.customName__c || customName__m"

>                  label="客户姓名"

>                 placeholder="请输入客户姓名"

> />
// customName__m为我在data里面定义的一个变量，接口赋值在给ruleForm.customName__c 赋值的时候会同时给customName__m赋值，
// 此时ios14+版本的iphone的输入框中能够正确显示出来值 ，请官方实测一下！如果是问题，请尽快修复！

**Comment by gaotianshuang** [comment]
您好，在使用van-picker 滑动时 滑动过快会定位不到日期 怎么解决～

**Comment by 13970883721** [comment]
card标签只填写整数价格的时候，会显示小数点。。。。  还有其他带金额的组件也是，必须用X.00格式的金额！不能根据用户给的参数显示吗。尴尬死了。

**Comment by Hui-Coder** [comment]
我记得，vant组件，但是有个对应的原型图设计来着，，，哪位大佬知道

**Reply by Hui-Coder** [reply]
当时

**Comment by sirT2626** [comment]
大佬们，建议增加一个视频播放的组件~

**Comment by zongyi1996** [comment]
使用vant的progress进度条组件，使用this.$refs.progress.resize()重绘方法不起作用

**Comment by 13782199098** [comment]
使用axios封装toast组件，请求拦截中加入toast.loading，相应拦截中toast.clear，但是请求是异步的，加载快的请求会把未请求完成的loading清除掉

**Comment by XieXianPing** [comment]
vant-cli创建的组件库，在哪里import自定义的js文件呢

**Comment by Lulinx** [comment]
我用webpack4搭了一个vue多页面框架，安装其他的UI组件库可以，vant就报错，不渲染，为啥啊

**Comment by Cifferni** [comment]
NoticeBar 文字过多会造成抖动，卡顿问题

**Comment by SuMasaki** [comment]
NoticeBar文字太多超过屏幕宽度会出现底部横向滚动条

**Reply by SuMasaki** [reply]
只在手机上才会显示

**Reply by SuMasaki** [reply]
苹果手机会有这个情况出现，安卓不会出现

**Comment by zgcwf** [comment]
steps 竖向步骤条 自定义图标时， 连接的竖线过长深入到图标内部，有点丑

**Comment by programmerlld** [comment]
vue2 使用checkbox，当选项多余两个，选择的result，返回值不对
![Uploading QQ截图20221027141128.png…]()

**Comment by programmerlld** [comment]
<van-checkbox-group v-model="result">
  <van-cell-group>
    <van-cell
      v-for="(item, index) in list"
      clickable
      :key="item"
      :title="`复选框 ${item}`"
      @click="toggle(index)"
    >
      <template #right-icon>
        <van-checkbox :name="item" ref="checkboxes" />
      </template>
    </van-cell>
  </van-cell-group>
</van-checkbox-group>

**Comment by xyj1020** [comment]
在vue3中使用vant， 用ts 定义了columns-type, 但是npm run build 的时候出现了以下的问题，求指教～

错误信息：
S2322: Type 'string[]' is not assignable to type 'TimePickerColumnType[] & TimePickerColumnType[]'.
  Type 'string[]' is not assignable to type 'TimePickerColumnType[]'.
    Type 'string' is not assignable to type 'TimePickerColumnType'.

17                             :columns-type="timeColumnsType"

       
    The expected type comes from property 'columnsType' which is declared here on type 'ComponentProps<ComponentPublicInstanceConstructor<{ $: ComponentInternalInstance; $data: {}; $props: Partial<{ formatter: (type: string, option: PickerOption) => PickerOption; ... 10 more ...; columnsType: DatePickerColumnType[]; }> & Omit<...>; ... 10 more ...; $watch<T extends string | ((...args: any) => any)>(sou...'



源代码：
let dateColumnsType: Ref<string[]> = ref(['year', 'month', 'day']);
let timeColumnsType: Ref<Array<string>> = ref(['hour', 'minute', 'second']);

<van-date-picker
                            :min-date="minDate"
                            :columns-type="dateColumnsType"
                            :confirm-button-text="showTimePicker ? '下一步' : '确认'"
                            @confirm="handleDateConfirm"
                            @cancel="handleCancel"
                        />

**Reply by xiejun-net** [reply]
看提示 
`Type 'string[]' is not assignable to type 'TimePickerColumnType[]'.`

vant 中 type DatePickerColumnType = 'year' | 'month' | 'day';

导出 type TimePickerColumnType 然后`let dateColumnsType: Ref<TimePickerColumnType[]> = ref(['year', 'month', 'day']);`就可以吧

**Comment by turn-back** [comment]
vant 2.0中单元测试中，如果src下存在vue文件，单元测试框架无法测试该文件？

**Comment by xw-xw12** [comment]
现在的vant4是不是还没有适配app

**Reply by xudehong123** [reply]
目前没有提供 CDN 形式的语言包，可以手动拷贝语言包的内容来使用。  如何手动拷贝呢

**Comment by xudehong123** [comment]
目前没有提供 CDN 形式的语言包，可以手动拷贝语言包的内容来使用。请问如何手动拷贝呢

**Comment by 123piggy3** [comment]
打包引入js文件使用报错  Unhandled error during execution of setup function 

<van-form @submit="onSubmit">
    <van-cell-group inset>
      <van-field v-model="username" @update:model-value="update" label="文本" />
      <div style="margin: 16px;">
        <van-button round block type="primary" native-type="submit">
          提交
        </van-button>
      </div>
    </van-cell-group>
  </van-form>

**Comment by zazahui-hugh** [comment]
showImagePreview函数调用时预览正常，滑动切换时图片索引改变了但是图片没有改变；组件使用时正常。
